src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -253,10 +255,15 @@ impl TargetInfo {
 
         // See rust-lang/cargo#4535.
         if target_triple.starts_with("wasm32-") && crate_type == "bin" && suffix == ".js" {
+            let flavor = if target_triple.contains("emscripten") {
```

Comment:
I think it's fine to skip the `emscripten` check here, I think wasm32 targets that primarily emit a js file are always emscripten targets